components/canvas/webgl_thread.rs

```diff
@@ -189,13 +192,32 @@ impl<VR: WebVRRenderHandler + 'static> WebGLThread<VR> {
         false
     }
 
+    /// Swap the underlying IOsurfaces upon requestAnimationFrame is called
+    fn swap_draw_buffers(&mut self, sender: WebGLSender<()>) {
+        for (context_id, ref mut data) in self.contexts.iter_mut() {
```

Comment:
It would be nice if we could avoid using two different mechanisms (triple-buffering on MacOS and GL sync'ing on other OSs). Is there not a way to use the existing locking API with IOSurfaces?